### PR TITLE
Post milestone1 fixes

### PR DESCRIPTION
**Pull request by Michal-MK**

### CODE REVIEW

**Comment by Michal-MK** [comment]
Merge only after you fix the entities you were responsible for in the first milestone. You can use this brach.

File: dao/src/main/java/cz/idomatojde/dao/common/BaseDAO.java [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 /** Base interface for all DAOs
  * @author Michal Hazdra
  * @param <TEntity> Any entity
  */
-public interface BaseDAO<TEntity> {
+public interface BaseDAO<TEntity extends IEntity> {
 
 
     /** Common functionality to persist a new entity
      * @param entity the entity to persist
```
Comment (JirkaVrbka) [inline dao/src/main/java/cz/idomatojde/dao/common/BaseDAO.java:19]: missing return doc

File: dao/src/test/java/cz/idomatojde/TestObjects.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package cz.idomatojde;
+
+import cz.idomatojde.entity.Category;
+import cz.idomatojde.entity.Offer;
+import cz.idomatojde.entity.User;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+
+public class TestObjects {
```
Comment (JirkaVrbka) [inline dao/src/test/java/cz/idomatojde/TestObjects.java:10]: TestObjectFactory maybe? To represent functionality

Comment (Michal-MK) [reply dao/src/test/java/cz/idomatojde/TestObjects.java:10]: "In Factory pattern, we create object without exposing the creation logic to the client and refer to newly created object using a common interface."
I think of these as "constant" fields in a static class. But classes in Java cannot be static ;)

Comment (JirkaVrbka) [reply dao/src/test/java/cz/idomatojde/TestObjects.java:10]: I was just referring to the name of the class

Comment (JirkaVrbka) [reply dao/src/test/java/cz/idomatojde/TestObjects.java:10]: After call, this is ok

**Review by JirkaVrbka — APPROVED** [review]

**Pull request merged**